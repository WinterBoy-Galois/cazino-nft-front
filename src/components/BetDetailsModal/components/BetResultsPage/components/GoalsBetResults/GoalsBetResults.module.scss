@import '../../../../../../styles/shared';

.container {
  width: 100%;

  @include media-breakpoint-up(md) {
    max-width: 340px;
  }
}

.list {
  padding: 0;
  margin: 0;
  list-style-type: none;

  &__item {
    display: flex;
    background-color: $blue;
    margin: 0.6rem 0;
    border-radius: 15px;

    &:first-of-type,
    &:last-of-type {
      margin: 0;
    }

    &--highlight {
      border: 2px solid $white;
    }
  }
}

.step {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 52px;
  border-radius: 15px;
  background-color: $light-blue;
  font-weight: 300;
  flex-shrink: 0;
}

.goals {
  display: flex;
  align-items: center;
  justify-content: space-around;
  padding: 0.5rem;
  width: 100%;
}

.goal {
  display: flex;
  align-items: center;

  &__index {
    width: 21px;
    height: 21px;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 50%;
    background-color: rgba($white, 0.5);
    color: #000;
    font-size: 16px;
    font-weight: 300;
  }

  &__ball {
    width: 29px;
    height: 29px;
    border: 2px solid $negative;
    border-radius: 50%;
    opacity: 0.5;

    &--win {
      border-color: $positive;
    }

    &--selected {
      opacity: 1;
    }

    &__icon {
      vertical-align: unset;
    }
  }
}
