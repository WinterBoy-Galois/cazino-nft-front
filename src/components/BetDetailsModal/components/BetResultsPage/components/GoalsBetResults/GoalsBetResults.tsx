import React, { useMemo } from 'react';
import styles from './GoalsBetResults.module.scss';
import { GoalSelection } from '../../../../../../models/betDetails.model';
import { isNullOrUndefined } from 'util';
import goal from '../../../../../../assets/images/games/goals/game-goals-icon.svg';

interface IProps {
  selections: GoalSelection[];
}

const GoalsBetResults: React.FC<IProps> = ({ selections }) => {
  selections = useMemo(() => selections.slice().sort((s1, s2) => s2.step - s1.step), [selections]);

  const getHighlightClass = (selection: GoalSelection, index: number) => {
    return (!isNullOrUndefined(selection.selected) &&
      selections[index - 1] &&
      isNullOrUndefined(selections[index - 1].selected)) ||
      (selections.filter(s => s.selected === undefined || s.selected === null).length <= 0 &&
        index === 0)
      ? styles['list__item--highlight']
      : '';
  };

  return (
    <div className={styles.container}>
      <ul className={styles.list}>
        {selections.map((s, i) => (
          <li
            key={`${s.step}_${s.luckySpots.toString()}_${s.selected}`}
            className={`${styles.list__item} ${getHighlightClass(s, i)}`}
          >
            <div className={styles.step}>{s.step + 1}</div>
            <div className={styles.goals}>
              {Array.from(new Array(3)).map((_, i) => (
                <div key={`goal_${i}`} className={styles.goal}>
                  <div className={styles.goal__index}>{i}</div>
                  <div
                    className={`${styles.goal__ball} ${
                      s.luckySpots.includes(i) ? styles['goal__ball--win'] : ''
                    }
                    ${s.selected === i ? styles['goal__ball--selected'] : ''}`}
                  >
                    <img className={styles.goal__ball__icon} src={goal} alt="Goal" />
                  </div>
                </div>
              ))}
            </div>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default GoalsBetResults;
