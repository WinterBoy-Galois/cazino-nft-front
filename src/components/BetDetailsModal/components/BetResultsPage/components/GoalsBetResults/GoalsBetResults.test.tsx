import React from 'react';
import { render } from '@testing-library/react';
import GoalsBetResults from '.';

describe('GoalsBetResults', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(
      <GoalsBetResults selections={[{ step: 0, luckySpots: [0, 2], selected: 0 }]} />
    );

    // Assert
    expect(container).toMatchSnapshot();
  });
});
