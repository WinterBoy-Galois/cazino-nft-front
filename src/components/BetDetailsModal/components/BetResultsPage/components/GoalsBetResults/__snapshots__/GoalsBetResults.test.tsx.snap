// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalsBetResults should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <ul
          class="list"
        >
          <li
            class="list__item list__item--highlight"
          >
            <div
              class="step"
            >
              1
            </div>
            <div
              class="goals"
            >
              <div
                class="goal"
              >
                <div
                  class="goal__index"
                >
                  0
                </div>
                <div
                  class="goal__ball goal__ball--win
                    goal__ball--selected"
                >
                  <img
                    alt="Goal"
                    class="goal__ball__icon"
                    src="game-goals-icon.svg"
                  />
                </div>
              </div>
              <div
                class="goal"
              >
                <div
                  class="goal__index"
                >
                  1
                </div>
                <div
                  class="goal__ball 
                    "
                >
                  <img
                    alt="Goal"
                    class="goal__ball__icon"
                    src="game-goals-icon.svg"
                  />
                </div>
              </div>
              <div
                class="goal"
              >
                <div
                  class="goal__index"
                >
                  2
                </div>
                <div
                  class="goal__ball goal__ball--win
                    "
                >
                  <img
                    alt="Goal"
                    class="goal__ball__icon"
                    src="game-goals-icon.svg"
                  />
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <ul
        class="list"
      >
        <li
          class="list__item list__item--highlight"
        >
          <div
            class="step"
          >
            1
          </div>
          <div
            class="goals"
          >
            <div
              class="goal"
            >
              <div
                class="goal__index"
              >
                0
              </div>
              <div
                class="goal__ball goal__ball--win
                    goal__ball--selected"
              >
                <img
                  alt="Goal"
                  class="goal__ball__icon"
                  src="game-goals-icon.svg"
                />
              </div>
            </div>
            <div
              class="goal"
            >
              <div
                class="goal__index"
              >
                1
              </div>
              <div
                class="goal__ball 
                    "
              >
                <img
                  alt="Goal"
                  class="goal__ball__icon"
                  src="game-goals-icon.svg"
                />
              </div>
            </div>
            <div
              class="goal"
            >
              <div
                class="goal__index"
              >
                2
              </div>
              <div
                class="goal__ball goal__ball--win
                    "
              >
                <img
                  alt="Goal"
                  class="goal__ball__icon"
                  src="game-goals-icon.svg"
                />
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
